testdata/buildpack/bin/detect.bat

```diff
@@ -0,0 +1,2 @@
+@echo off
+bash %~dp0\detect %1 %2
```

Comment:
I was surprised to find that I couldn't run the detect unit tests on a windows machine w/o `bash` installed. The failures give the false impression that `bash` is a dependency of the detector. However the implementation is correctly calling `detect.bat` which in turn requires `bash`. Given that we generally won't have `bash` available when the lifecycle runs on windows I am not sure I am comfortable with the dependency in the test. It makes it impossible to validate that the lifecycle can execute correctly in an environment sans `bash`. Can we reimplement the detect test logic in the `*.bat` files?

Reply:
We can rewrite these in pure `batch`. The makefile provides a way to run the tests in a docker container that has all the necessary dependencies, but I understand the overall concerns.